### PR TITLE
Fixes the charm registration message.

### PR DESCRIPTION
**Pull request by alesstimec**
Improve the charm registration error message. If there is no option make that clear, rather than present a list with no entries.

### CODE REVIEW

**Comment by mattyw** [comment]
Please add a description for this change

File: cmd/juju/application/register.go [hunk 1]
```diff
@@ -76,7 +76,11 @@ func (r *RegisterMeteredCharm) RunPre(api MeteredDeployAPI, bakeryClient *httpba
 					return err1
 				}
 				charmURL := deployInfo.CharmID.URL.String()
-				return errors.Errorf(`%v has no default plan. Try "juju deploy --plan <plan-name> with one of %v"`, charmURL, strings.Join(options, ", "))
+				if len(options) > 0 {
```
Comment (mattyw) [inline cmd/juju/application/register.go]: Just remove some indentation
```
if len(options) == 0 {
   return errors.Errorf("no plans available for %v.", charmURL)
}
return errors.Errorf(`%v has no default plan. Try "juju deploy --plan <plan-name> with one of %v"`, charmURL, strings.Join(options, ", "))
```

File: cmd/juju/application/register_test.go [hunk 2]
```diff
@@ -417,6 +423,37 @@ func (s *registrationSuite) TestMeteredCharmNoDefaultPlan(c *gc.C) {
 	}})
 }
 
+func (s *registrationSuite) TestMeteredCharmNoAvailablePlan(c *gc.C) {
```
Comment (axw) [inline cmd/juju/application/register_test.go]: maybe also a test when there are options?

Comment (alesstimec) [inline cmd/juju/application/register_test.go]: @axw there is one: TestMeteredCharmNoDefaultPlan

**Comment by alesstimec** [comment]
!!build!! !!rebuild!!

**Comment by alesstimec** [comment]
$$merge$$

**Pull request merged**